pulp_container/app/registry_api.py

```diff
@@ -947,8 +914,10 @@ def put(self, request, path, pk=None):
 
             config_digest = config_layer.get("digest")
             try:
+                blobs_in_latest_version = repository.latest_version().content.values_list("pk")
+                pending_blobs = repository.pending_blobs.values_list("pk")
                 config_blob = models.Blob.objects.get(
-                    digest=config_digest, pk__in=repository.latest_version().content
+                    digest=config_digest, pk__in=chain(blobs_in_latest_version, pending_blobs)
```

Comment:
Chain will evaluate these queries instead of creating subqueries. I think it should be
```python
available_blobs = repository.latest_version().content || repository.pending_blobs
config_blob = ... get(..., pk__in=avaliable_blobs.values_list("pk"))
```

Reply:
You cannot do this in that way.

```
In [8]: Blob.objects.filter(pk__in=repo.pending_blobs || repo.latest_version().content)
  Cell In[8], line 1
    Blob.objects.filter(pk__in=repo.pending_blobs || repo.latest_version().content)
                                                   ^
SyntaxError: invalid syntax
```

```
----> 1 Blob.objects.filter(pk__in=repo.pending_blobs.all() | repo.latest_version().content)

File /usr/local/lib/python3.8/site-packages/django/db/models/query.py:345, in QuerySet.__or__(self, other)
    343 if not other.query.can_filter():
    344     other = other.model._base_manager.filter(pk__in=other.values('pk'))
--> 345 combined.query.combine(other.query, sql.OR)
    346 return combined

File /usr/local/lib/python3.8/site-packages/django/db/models/sql/query.py:586, in Query.combine(self, rhs, connector)
    577 def combine(self, rhs, connector):
    578     """
    579     Merge the 'rhs' query into the current one (with any 'rhs' effects
    580     being applied *after* (that is, "to the right of") anything in the
   (...)
    584     'rhs' query.
    585     """
--> 586     assert self.model == rhs.model, \
    587         "Cannot combine queries on two different base models."
    588     assert not self.is_sliced, \
    589         "Cannot combine queries once a slice has been taken."
    590     assert self.distinct == rhs.distinct, \
    591         "Cannot combine a unique query with a non-unique query."

AssertionError: Cannot combine queries on two different base models.
```

Reply:
Ha, this helped:
```
In [15]: repo.latest_version().content.values_list("pk").union(repo.pending_blobs.values_list("pk"))
```

Reply:
That sounds great!